unionfs/unionfs.go

```diff
@@ -712,7 +712,12 @@ func (fs *unionFS) GetXAttr(name string, attr string, context *fuse.Context) ([]
 	}
 	r := fs.getBranch(name)
 	if r.branch >= 0 {
-		return fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		value, code := fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		// We intercept ENOSYS and replace it with ENODATA; returning ENOSYS would cause FUSE to never to GetXAttr() again.
+		if code == fuse.ENOSYS {
+			return nil, fuse.ENODATA
+		}
+		return value, code
```

Comment:
It seems like either ENOSYS should be converted into ENODATA in opcode.go or that the approach in https://github.com/hanwen/go-fuse/commit/7b281480b21a59f54a0c86451e7de13cd9be736f should be continued (i.e. change ENOSYS to ENODATA in fuse/pathfs/default.go).

If we choose the former, then I'd suggest reverting the previous changes in fuse/defaultraw.go and fuse/nodefs/defaultnode.go.

Let me know.